### ISSUE TITLE
Timeline Resize Bug

### ISSUE THREAD

**Issue by ghost** [OpenShot/openshot-qt#5647]
**Describe the bug:**
When closing and opening OpenShot, the timeline resizes on its own, smushing the Timeline Player each time.

**Steps to reproduce the behavior:**
Open and close Openshot about 3 times on a Fresh Install.

**Expected behavior:**
The timeline resizing on its own should not resize upon repeated close and open of OpenShot

**System Details:**
- OpenShot Version [e.g. 2.4.3]: 3.2.1
- Operating System / Distro: Linux Mint 22 with Cinnamon DE 6.2.9

**Log Files:**
No crashing

**Exception / Stacktrace:**
Nothing weird about this issue

**Screenshots:** (Optional)

https://github.com/user-attachments/assets/a2036cc9-5e6e-414f-b5c8-7d9f9e8c07a1

**Comment by Colorjet3** [comment]
Not really seeing or understanding the issue.  You are going a bit fast with your recorder.

May be you can add voice to your recording explaining what exactly is happening.

**Comment by ghost** [comment]
Apologies, let me try again

https://github.com/user-attachments/assets/a8c0f77b-9af5-4726-9564-d42239c0cff5

**Comment by Colorjet3** [comment]
Thank you @BlueberryDingle for the new recording and the clear explanation.  My host computer is Winodws Pro 11 and the Linux Mint that I am running is in Oracle Virtualbox.

I will do some testing and get back to you with my results.

If you don't mind and give this a try and let me know the results as well:
1. Close OpenSHot if frunning. 
2. Go to Openshot.org/download and download the Appimage v3.2.1
3.  Make it executable.
4.  You might need to install libfuse2 for the AppImage to run:  sudo apt install libfuse2
5.  Run the AppImage and perform your test again.
6.  You don't need to record again now that I understand clearly what you are running into.  Just let me know if the problem persists or it is still there.
Thank you in advance!

**Comment by ghost** [comment]
> Thank you @BlueberryDingle for the new recording and the clear explanation. My host computer is Winodws Pro 11 and the Linux Mint that I am running is in Oracle Virtualbox.
> 
> I will do some testing and get back to you with my results.
> 
> If you don't mind and give this a try and let me know the results as well:
> 
>     1. Close OpenSHot if frunning.
> 
>     2. Go to Openshot.org/download and download the Appimage v3.2.1
> 
>     3. Make it executable.
> 
>     4. You might need to install libfuse2 for the AppImage to run:  sudo apt install libfuse2
> 
>     5. Run the AppImage and perform your test again.
> 
>     6. You don't need to record again now that I understand clearly what you are running into.  Just let me know if the problem persists or it is still there.
>        Thank you in advance!

No problem. Its interesting you mentioned the steps to reproduce the issue. I did infact completely re-download OpenShot and made sure to delete the .openshot_qt folder as I noticed how blurry the timeline player was under the "Disk" method for caching, I noticed the same resize bug occured on the fresh install.

I tried swapping themes which did not help, I did unfreeze the view, positioned the tabs to my liking and then freezing the view, to no avail the resize bug kept happening.

**Comment by Colorjet3** [comment]
Hello @BlueberryDingle 

I test in both Windows 11 Pro (my host device) and Linux Mint 21.3 (guest virtualbox) and I am unable to reproduce the issue.  In 1.  Windows 11 Pro I am running the latest dev daily build #12980.
2.  In Linux Mint, I tested with both the production release v3.2.1 and the latest dev daily build #12980 AppImage.

My Linux environment does not have direct access to my hardware (NVIDIA GeForce RTX 2080 card). 

3.  What are your computer specs?
a.  Is it a laptop or desktop?
b.  What is your CPU, GPU/RAM, RAM, Hard drive type?

4.  Cache settings.  Do not use "Disk" for "Cache Mode" in Edit | Preferences | Cache tab.  Stick with "Memory" instead.
5.  If you download the latest dev daily build AppImage (build #12980 as of this writing), you can reset your settings on the different tabs in Edit | Preferences to "Restore Defaults: xxxxxxx".  Recommended Cache settings are:

Cache Mode:                          =                Memory
Cache Pre-roll:: Min Frames   = 24          <--Change this to 48
Cache Pre-roll:: Max Frames   = 48         <--Change this to 96
Cache Ahead (Percent)           = .70         <--Don't change this
Cache Max Frames                 = 600        <-- Change this to 1024 (this is the max)
Cache Limit (MB)                   = 512         <-- Change this 2048.

Do no make the Cache Limit (MB) too large even if you have lots of RAM.  I have a 32GB system and I never go above 4096 even for larger/complex projects.  Now, if you have a really high end gaming/video editing computer then you might be able to get away with the Cache Limit to be higher.

I forgot to ask an important question:  Are you projects impacted by this issue/bug?

**Comment by ghost** [comment]
> Hello @BlueberryDingle
> 
> I test in both Windows 11 Pro (my host device) and Linux Mint 21.3 (guest virtualbox) and I am unable to reproduce the issue. In 1. Windows 11 Pro I am running the latest dev daily build #12980. 2. In Linux Mint, I tested with both the production release v3.2.1 and the latest dev daily build #12980 AppImage.
> 
> My Linux environment does not have direct access to my hardware (NVIDIA GeForce RTX 2080 card).
> 
>     3. What are your computer specs?
>        a.  Is it a laptop or desktop?
>        b.  What is your CPU, GPU/RAM, RAM, Hard drive type?
> 
>     4. Cache settings.  Do not use "Disk" for "Cache Mode" in Edit | Preferences | Cache tab.  Stick with "Memory" instead.
> 
>     5. If you download the latest dev daily build AppImage (build #12980 as of this writing), you can reset your settings on the different tabs in Edit | Preferences to "Restore Defaults: xxxxxxx".  Recommended Cache settings are:
> 
> 
> Cache Mode: = Memory Cache Pre-roll:: Min Frames = 24 <--Change this to 48 Cache Pre-roll:: Max Frames = 48 <--Change this to 96 Cache Ahead (Percent) = .70 <--Don't change this Cache Max Frames = 600 <-- Change this to 1024 (this is the max) Cache Limit (MB) = 512 <-- Change this 2048.
> 
> Do no make the Cache Limit (MB) too large even if you have lots of RAM. I have a 32GB system and I never go above 4096 even for larger/complex projects. Now, if you have a really high end gaming/video editing computer then you might be able to get away with the Cache Limit to be higher.
> 
> I forgot to ask an important question: Are you projects impacted by this issue/bug?

Greetings, thats odd that you are unable to reduce this bug. For my computer specs:
- CPU: Intel Core i5-11500 @ 4.60Ghz (according to `neofetch`)
- GPU: Intel UHD Graphics 750 RocketLake-S GT1 Integrated
- RAM: 32GB of DDR4 in Dual Channel
- PC Type: Desktop tower
- SSD type: Crucial BX Series with 480GB of capacity
- Projects are uneffected by this issue. I believe OpenShot is attempting to show all 5 Tracks in one view, when I only want 3 Tracks, sadly not possible yet.

**Comment by ghost** [comment]
So, upon testing the nightly build, I can confirm this issue also happens in the nightly build.

**Comment by Colorjet3** [comment]
I just tested in Ubuntu 24.04 lts running as a guest in my Oracle Virtualbox environment (running the latest dev daily build #12980 **AppImage**) and I can not reproduce the behavior in this environment either.

I am not sure where to go from here.  I am going to assign this to the lead developer to see if he has any ideas.

**Comment by ghost** [comment]
The terminal output did not give much in terms of how this bug is occuring.

However, what i think is causing it are these messages:
```
Property 'modal'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'windowModality'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'enabled'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'geometry'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'frameGeometry'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'normalGeometry'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'x'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'y'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'pos'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'frameSize'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'size'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'width'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'height'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'rect'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'childrenRect'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'childrenRegion'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'sizePolicy'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'minimumSize'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'maximumSize'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'minimumWidth'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'minimumHeight'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'maximumWidth'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'maximumHeight'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'sizeIncrement'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'baseSize'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'palette'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'font'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'cursor'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'mouseTracking'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'tabletTracking'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'isActiveWindow'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'focusPolicy'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'focus'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'contextMenuPolicy'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'updatesEnabled'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'visible'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'minimized'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'maximized'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'fullScreen'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'sizeHint'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'minimumSizeHint'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'acceptDrops'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'windowOpacity'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'windowModified'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'toolTip'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'toolTipDuration'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'statusTip'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'whatsThis'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'accessibleName'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'accessibleDescription'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'layoutDirection'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'autoFillBackground'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'styleSheet'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'locale'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'windowFilePath'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'inputMethodHints'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'title'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'url'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'selectedText'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'hasSelection'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
Property 'zoomFactor'' of object 'TimelineView' has no notify signal and is not constant, value updates in HTML will be broken!
```

Notice how "TimelineView" is constantly repeated. Perhaps that could lead to resolving this bug.

**Comment by ghost** [comment]
Somehow, OpenShot is not correctly storing X,Y,Z values set by the users correctly
